### PR TITLE
[Fix #763] Add rule suggesting not to use RUBY_VERSION in gemspec

### LINKED ISSUE #763: Do not use RUBY_VERSION in gemspec
Taken from https://github.com/rubocop-hq/rubocop/issues/7137

Gemspec should not contain `RUBY_VERSION` as a condition to switch dependencies.
For example:

```ruby
Gem::Specification.new do |s|
  if RUBY_VERSION >= '2.5'
    s.add_runtime_dependency 'gem_a'
  else
    s.add_runtime_dependency 'gem_b'
  end
end
```

It has unexpected behavior. The RUBY_VERSION is determined by `rake release`.
If I use Ruby 2.5 to execute `rake release`, the gem always has gem_a as a dependency even if a user installs the gem with Ruby 2.4 or older.

We have several ways to fix the problem.

* Remove the dependencies and use `post_install_message` to notice the gems.
* Add both gems to the dependency.
* If the dependencies are for development, move them to Gemfile.

@pocke Feel free to edit this.

### PR DESCRIPTION
**Pull request by tejasbubane**
Closes https://github.com/rubocop-hq/ruby-style-guide/issues/763

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -4644,6 +4644,31 @@ https://ruby-doc.com/docs/ProgrammingRuby/html/rdtool.html[here].
 RD predated the rise of RDoc and YARD and was effectively obsoleted by them.footnote:[According to this https://en.wikipedia.org/wiki/Ruby_Document_format[Wikipedia article] the format used to be popular until the early 2000s when it was superseded by RDoc.]
 ****
 
+== Gemfile & Gemspec
```
Comment (koic) [inline README.adoc]: Can you use `and` instead of `&`?

```diff
-== Gemfile & Gemspec
+== Gemfile and Gemspec
```

File: README.adoc [hunk 2]
```diff
@@ -4644,6 +4644,31 @@ https://ruby-doc.com/docs/ProgrammingRuby/html/rdtool.html[here].
...
+
+=== No RUBY_VERSION in gemspec [[no-ruby-version-gemspec]]
```
Comment (koic) [inline README.adoc]: Likely it is better to add `the` and backquotes.

```diff
-=== No RUBY_VERSION in gemspec [[no-ruby-version-gemspec]]
+=== No `RUBY_VERSION` in the gemspec [[no-ruby-version-gemspec]]
```

File: README.adoc [hunk 3]
```diff
@@ -4644,6 +4644,31 @@ https://ruby-doc.com/docs/ProgrammingRuby/html/rdtool.html[here].
...
+
+Gemspec should not contain `RUBY_VERSION` as a condition to switch dependencies.
```
Comment (koic) [inline README.adoc]: Ditto.

```diff
-Gemspec should not contain `RUBY_VERSION` as a condition to switch dependencies.
+The gemspec should not contain `RUBY_VERSION` as a condition to switch dependencies.
```

**Comment by tejasbubane** [comment]
@koic Fixed.

**Comment by koic** [comment]
Thanks!

**Pull request merged**